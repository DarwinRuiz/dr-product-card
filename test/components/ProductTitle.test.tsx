
import React from 'react'
import { render } from '@testing-library/react'
import { ProductTitle, } from '../../src/components/ProductTitle'
import { ProductCard } from '../../src/components/ProductCard'
import { product1 } from '../data/products'

describe('ProductTitle', () => {
    test('renders ProductTitle component width custom title succeded', () => {
        const { container } = render(<ProductTitle title="Custom title" />)

        expect(container).toMatchSnapshot()
    });

    test('Should show the component with product name', () => {
        const { container } = render(<ProductCard product={product1}>
            {
                () => (
                    <ProductTitle />
                )
            }
        </ProductCard>)

        expect(container).toMatchSnapshot()
    });
})