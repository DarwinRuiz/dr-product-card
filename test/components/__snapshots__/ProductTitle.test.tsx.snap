// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Should show the component with product name 1`] = `
<div>
  <div
    class="productCard undefined"
  >
    <span
      class="productDescription undefined"
    >
      Coffee Mug - Con Imagen
    </span>
  </div>
</div>
`;

exports[`ProductTitle renders ProductTitle component width custom title succeded 1`] = `
<div>
  <span
    class="productDescription undefined"
  >
    Custom title
  </span>
</div>
`;
